src/main/java/io/vertx/core/http/HttpServerRequest.java

```diff
@@ -170,6 +169,18 @@ default String getParam(String paramName) {
     return params().get(paramName);
   }
 
+  /**
+   * Return the first param value or defaultValue if the query param is not provided.
+   *
+   * @param paramName    the param name
+   * @param defaultValue the default value, must be non-null
+   * @return the param value or default value
+   */
+  default String getParam(String paramName, String defaultValue) {
+    Objects.requireNonNull(defaultValue, "defaultValue");
+    return params().contains(paramName) ? params().get(paramName) : defaultValue;
```

Comment:
This method is a good idea! 😎  I know I'm nitpicking, but may I suggest to skip the `contains()` call and just do a `get()` and if `null`, then return `defaultValue`?  This will skip at least one MultiMap call.

Reply:
> I know I'm nitpicking

Noo, I was actually thinking about this way of implementation too, but I found `contains()` then `get()` more general and interface-friendlier.

But you're right as definition of `MultiMap#get()` is: _Something or **null** if there is no such entry_. 
Re-implemented as you have suggested, thanks.